test/conformance/api/v1/util.go

```diff
@@ -121,14 +121,28 @@ func validateDataPlane(t pkgTest.TLegacy, clients *test.Clients, names test.Reso
 // Validates the state of Configuration, Revision, and Route objects for a runLatest Service.
 // The checks in this method should be able to be performed at any point in a
 // runLatest Service's lifecycle so long as the service is in a "Ready" state.
-func validateControlPlane(t *testing.T, clients *test.Clients, names test.ResourceNames, expectedGeneration string) error {
+func validateControlPlane(t *testing.T, clients *test.Clients, names test.ResourceNames, expectedGeneration string, imagesForMultipleContainers ...map[string]string) error {
 	t.Log("Checking to ensure Revision is in desired state with", "generation", expectedGeneration)
 	err := v1test.CheckRevisionState(clients.ServingClient, names.Revision, func(r *v1.Revision) (bool, error) {
 		if ready, err := v1test.IsRevisionReady(r); !ready {
 			return false, fmt.Errorf("revision %s did not become ready to serve traffic: %w", names.Revision, err)
 		}
-		if validDigest, err := shared.ValidateImageDigest(t, names.Image, r.Status.DeprecatedImageDigest); !validDigest {
-			return false, fmt.Errorf("imageDigest %s is not valid for imageName %s: %w", r.Status.DeprecatedImageDigest, names.Image, err)
+		// for multi containers there will be more than one container and names.Image is a string,
+		// so in order to validate imageDigest for each images validateControlPlane accepting optional field called imagesForMultipleContainers
+		if names.Image == "" {
+			for _, value := range imagesForMultipleContainers {
```

Comment:
I think this loop is not necessary as there should always only be one map passed. Instead, this might want to do something like this

```golang
if len(imagesForMultipleContainers) > 0 {
  images := imagesForMultipleContainers[0]
} else {
  // Single container case
}
```

That should spare you 2 levels of indentation.

Though I guess this is more what @julz was getting at above too. If we added the images through the names struct we'd spare the entire extra parameter. No strong opinion on that yet though.

Reply:
Hey @markusthoemmes good suggestion

Planning to handle this and @julz comment together as a follow up PR

Will that be okay?

Reply:
wfm